pydantic_ai_slim/pydantic_ai/models/groq.py

```diff
@@ -545,9 +545,14 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
                 except IndexError:
                     continue
 
+                provider_details_dict: dict[str, Any] = {}
                 if raw_finish_reason := choice.finish_reason:
-                    self.provider_details = {'finish_reason': raw_finish_reason}
+                    provider_details_dict['finish_reason'] = raw_finish_reason
                     self.finish_reason = _FINISH_REASON_MAP.get(raw_finish_reason)
+                if self._provider_timestamp is not None:  # pragma: no branch
+                    provider_details_dict['timestamp'] = self._provider_timestamp
```

Comment:
Same as above; we're unnecessarily rebuilding the entire provider details list on each iteration